### PR TITLE
fix batch insert

### PR DESCRIPTION
**Pull request by fabienjuif**
Hi

When we set the `batchSize` option, the stream never ends.
This is because the first resolve (happening on the first `write` event) never resolves the Promise.

This PR fix that!

Thank you to share your work @AbdullahAli !

### CODE REVIEW

**Comment by fabienjuif** [comment]
I dunno why tests doesn't work on travis, it works on my local machine.
Can I let you debug that ?

I publish this version on this: https://www.npmjs.com/package/stream-to-mongo-db-fabienjuif
Waiting for your merge :)

**Comment by fabienjuif** [comment]
My mistake @KainHaart , this is fixed 👍

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
-  "name": "stream-to-mongo-db",
-  "version": "1.1.0",
+  "name": "stream-to-mongo-db-fabienjuif",
```
Comment (AbdullahAli) [inline package.json]: Can we change this please @fabienjuif

Comment (fabienjuif) [reply package.json]: Haaa @AbdullahAli just in time :)

I was going to make it a fork.
I change that, as you saw I fix error handling to :)

Comment (AbdullahAli) [reply package.json]: Thank you for the work, there are some failing tests, can you please take a look?

**Comment by fabienjuif** [comment]
@AbdullahAli I fix the name back.

Do you think to publish it soon ?

File: package.json (hunk 1 above)
Comment (fabienjuif) [reply package.json]: I have the feeling that tests are not concurrent safe.
I am not sure how to fix that but rewriting them :/

Comment (AbdullahAli) [reply package.json]: Bear with me, I will do a rewrite to simplify the codebase

Comment (fabienjuif) [reply package.json]: @AbdullahAli I fix tests in a new commit if you want to see that

**Comment by fabienjuif** [comment]
@AbdullahAli tests are simplified and fixed 🎉

**Comment by AbdullahAli** [comment]
@fabienjuif great stuff! ⭐️

**Comment by fabienjuif** [comment]
thank you @AbdullahAli for the initial work and your merge :)
Do you think you will publish a new version soon so I can reference back your lib (instead of my fork) and my client would be happy 😜

**Comment by AbdullahAli** [comment]
I just published it, all done

thank you for your work 👍

**Comment by fabienjuif** [comment]
Thank you :)

**Pull request merged**